### PR TITLE
24 endpoints get lessons by dates

### PR DESCRIPTION
**Pull request by KaterynaHerasymiva**

### CODE REVIEW

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 1]
```diff
@@ -45,4 +45,24 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
...
+        var dayCards = await _context.Lessons
+            .Where(c => c.StartAt.Date >= requestDto.Date.Date
+                        && c.StartAt.Date <= requestDto.Date.AddDays(6).Date
+                        )
+            .GroupBy(c => c.StartAt.Date)
+            .Select(t =>
+                new DayCardDto()
+                {
+                    Date = t.Key,
+                    DayOfWeek = t.Key.DayOfWeek,
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:60]: Not needed, you send this in a model

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 2]
```diff
@@ -45,4 +45,24 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
 
         return lessonDtos;
     }
-}
+
+    public async Task<ICollection<DayCardDto>?> GetDayCardsOfWeekAsync(RequestDayCardDto requestDto)
+    {
+        var dayCards = await _context.Lessons
+            .Where(c => c.StartAt.Date >= requestDto.Date.Date
+                        && c.StartAt.Date <= requestDto.Date.AddDays(6).Date
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:53]: Can you make sure that the date you use will always be Monday? Independent of what you get in request
Just get the first Monday before that date

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs (hunk 1 above)
Comment (KaterynaHerasymiva) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs]: fixed

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs (hunk 2 above)
Comment (KaterynaHerasymiva) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs]: fixed

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 3]
```diff
@@ -45,4 +45,24 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
 
         return lessonDtos;
     }
-}
+
+    public async Task<ICollection<DayCardDto>?> GetDayCardsOfWeekAsync(RequestDayCardDto requestDto)
+    {
+
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:51]: new line

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by Anatolii-Ivanov — DISMISSED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 4]
```diff
@@ -47,6 +47,26 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
         return lessonDtos;
     }
 
+    public async Task<ICollection<DayCardDto>?> GetDayCardsOfWeekAsync(RequestDayCardDto requestDto)
+    {
+        var mondayDate = requestDto.Date.AddDays(-(int)requestDto.Date.DayOfWeek).Date;
+        var dayCards = await _context.Lessons
+            .Where(c => c.StartAt.Date >= mondayDate
+                        && c.StartAt.Date <= mondayDate.AddDays(6)
+                        )
```
Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:56]: Please do not use magic numbers

Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:56]:
```suggestion
            .Where(c => c.StartAt.Date >= mondayDate 
                && c.StartAt.Date <= mondayDate.AddDays(6))
```

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs [hunk 5]
```diff
@@ -24,6 +24,13 @@ public async Task<ActionResult<ICollection<LessonDto>>> GetAllAsync([FromBody] F
             return Ok(lessons);
         }
 
+        [HttpGet("week")]
+        public async Task<ActionResult<ICollection<DayCardDto>>> GetDayCardAsync([FromQuery] RequestDayCardDto requestDto)
+        {
+            var lessons = await _lessonsService.GetDayCardsOfWeekAsync(requestDto);
+            return Ok(lessons);
+        }
```
Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs:32]:
```suggestion
        public async Task<ActionResult<ICollection<DayCardDto>>> GetDayCardAsync([FromQuery] RequestDayCardDto requestDto) => _lessonsService.GetDayCardsOfWeekAsync(requestDto);
```

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs (hunk 4 above)
Comment (KaterynaHerasymiva) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:56]: fixed

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs (hunk 5 above)
Comment (KaterynaHerasymiva) [reply backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs:32]: This change requires also method signature change like
`[HttpGet("week")]
        public async Task<ICollection<DayCardDto>> GetDayCardAsync([FromQuery] RequestDayCardDto requestDto)
            => (await _lessonsService.GetDayCardsOfWeekAsync(requestDto))!;`
Should we change all the methods in controller in the same way?

Comment (KaterynaHerasymiva) [reply backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs:32]: fixed

**Review by Anatolii-Ivanov — DISMISSED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs [hunk 6]
```diff
@@ -20,10 +20,13 @@ public LessonsController(ILessonsService lessonsService)
...
         {
             var lessons = await _lessonsService.GetAllLessonsAsync(filtersRequest);
-
             return Ok(lessons);
         }
 
+        [HttpGet("week")]
+        public async Task<ICollection<DayCardDto>> GetDayCardAsync([FromQuery] RequestDayCardDto requestDto)
+            => (await _lessonsService.GetDayCardsOfWeekAsync(requestDto))!;
+
```
Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs:29]: remove async and and await

Comment (KaterynaHerasymiva) [reply backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/LessonsController.cs]: fixed

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**